### ISSUE TITLE
CROSSCOMPILE on 0.26.0 for ARM returns with no matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type`

### ISSUE THREAD

**Issue by henryqinshield** [bazelbuild/bazel#8631]
> ATTENTION! Please read and follow:
> - if this is a _question_ about how to build / test / query / deploy using Bazel, ask it on StackOverflow instead: https://stackoverflow.com/questions/tagged/bazel
> - if this is a _discussion starter_, send it to bazel-discuss@googlegroups.com
> - if this is a _bug_ or _feature request_, fill the form below as best as you can.

### Description of the problem / feature request:

We are trying to crosscompile our targets using bazel 0.26.1 for ARM. Everything works when we try to run using k8, however, when we switch to `--cpu=arm` we get the following error:

```
WARNING: Ignoring JAVA_HOME, because it must point to a JDK, not a JRE.
INFO: Invocation ID: 61c08ab3-983f-474e-bb68-05e3a49158e9
INFO: ToolchainResolution: Selected execution platform @bazel_tools//platforms:host_platform, 
INFO: ToolchainResolution: Looking for toolchain of type @bazel_tools//tools/cpp:toolchain_type...
INFO: ToolchainResolution:   Considering toolchain @local_config_cc//:cc-compiler-armabi-v7a...
INFO: ToolchainResolution:     Toolchain constraint @bazel_tools//platforms:os has value @bazel_tools//platforms:android, which does not match value @bazel_tools//platforms:linux from the target platform @bazel_tools//platforms:target_platform
INFO: ToolchainResolution:   Rejected toolchain @local_config_cc//:cc-compiler-armabi-v7a, because of target platform mismatch
INFO: ToolchainResolution:   Considering toolchain @local_config_cc//:cc-compiler-k8...
INFO: ToolchainResolution:     Toolchain constraint @bazel_tools//platforms:cpu has value @bazel_tools//platforms:x86_64, which does not match value @bazel_tools//platforms:arm from the target platform @bazel_tools//platforms:target_platform
INFO: ToolchainResolution:   Rejected toolchain @local_config_cc//:cc-compiler-k8, because of target platform mismatch
INFO: ToolchainResolution:   No toolchains found
ERROR: While resolving toolchains for target @org_lzma_lzma//:lzma: no matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type
ERROR: Analysis of target '@org_lzma_lzma//:lzma' failed; build aborted: no matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Setting --cpu=arm causes the issue to manifest

### What operating system are you running Bazel on?

Ubuntu 16.04

### What's the output of `bazel info release`?

WARNING: Ignoring JAVA_HOME, because it must point to a JDK, not a JRE.
INFO: Invocation ID: de5b36b8-0198-4894-9c66-a72299d652f1
release 0.26.0

**Comment by aiuto** [comment]
@katre @hlopko

**Comment by katre** [comment]
Do you have a sample repo I can clone and demonstrate the problem with?

**Comment by hlopko** [comment]
Do you provide your own arm toolchain? Bazel's autoconfigured C++ toolchain at @local_config_cc only works for host compilation (so no cross compiling), and the generated arm toolchain in @local_config_cc is only a placeholder toolchain for android rules, so it cannot build anything anyway.

There are some cross-compilation toolchains for Bazel on the Github, but I'm not sure how well they work. If you want to pursue writing your own, see https://docs.bazel.build/versions/0.26.0/cc-toolchain-config-reference.html and https://docs.bazel.build/versions/0.26.0/tutorial/cc-toolchain-config.html.

**Comment by hlopko** [comment]
To double check, this never worked for you right? I want to make sure it's not a Bazel upgrade issue.

**Comment by henryqinshield** [comment]
I was able to get this to build over the weekend. We were using our own arm toolchain. This had worked in 0.21.0 but during the upgrade to 0.26, it broke. We had to register create a toolchain from our cc_toolchain and then register it in our WORKSPACE. The documentation was a little bit hard to follow since I didn't see anything in the tutorial about registering a toolchain in the WORKSPACE. But we managed to get it to work!
